### ISSUE TITLE
[BITV] 9.4.1.2/3.8 - The tables (including table in the dialog "Choose target folder") are sortable, but the sorting status is only visible to visual users. (2)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/server#37105]
One possibility to make the sorting status available to users of assistive technologies would be to use the aria-sort attribute. Additionally, the implementation of sorting buttons using links is incorrect. Button elements should be used instead. 

See https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/ for reference.

### Files
- [x] https://github.com/nextcloud/server/pull/40893

### File picker
- [x] `aria-sort`
- [x] https://github.com/nextcloud-libraries/nextcloud-dialogs/pull/1058
- [x] Release
- [x] https://github.com/nextcloud/server/pull/40956

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v3-n8

</details>

- [x] 9.4.1.2/10.8 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-bfbf07e5c3-v10-n8

**Comment by Pytal** [comment]
Our current implementation now uses buttons like below
```
th
  button aria-label="Sort list by Name (ascending)"
th
```

With the sort order being within the label instead of a separate `aria-sort`

I suppose the ideal implementation would use `aria-sort` & `aria-pressed` as described in https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort @michaelnissenbaum ?

**Comment by AndyScherzinger** [comment]
Hi @michaelnissenbaum - any feedback on https://github.com/nextcloud/server/issues/37105#issuecomment-1703389457 ? Thanks in advance

**Comment by michaelnissenbaum** [comment]
I would also prefer the solution described at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-sort or https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/. The use of the "aria-pressed" attribute is not necessary. Using only "aria-sort" on the "th" element is sufficient. Also, take a look at the article https://adrianroselli.com/2021/04/sortable-table-columns.html - "If your column is not sorted, instead of using aria-sort="none" remove the attribute instead. You should not use aria-sort on more than one column header at a time.".

**Comment by Pytal** [comment]
Is dialogs lib ready for another beta @susnux ?

**Comment by susnux** [comment]
@Pytal released feel free to update server
